src/components/tx/Transactions.js

```diff
@@ -32,233 +32,263 @@ import PaginationURL from '../../utils/pagination';
-                ref="txNext"
-                className={
-                  !this.state.hasAfter || this.state.transactions.length === 0
-                    ? 'page-item disabled'
-                    : 'page-item'
-                }
-              >
-                <Link
-                  className="page-link"
-                  to={this.pagination.setURLParameters({
-                    ts: this.state.lastTimestamp,
-                    hash: this.state.lastHash,
-                    page: 'next',
-                  })}
-                >
-                  Next
-                </Link>
-              </li>
-            </ul>
-          </nav>
-        );
+        // First load without parameters
+        newHasBefore = false;
+        newHasAfter = data.has_more;
       }
-    };
 
-    const loadTable = () => {
-      return (
-        <div className="table-responsive mt-5">
-          <table className="table table-striped" id="tx-table">
-            <thead>
-              <tr>
-                <th className="d-none d-lg-table-cell">Hash</th>
-                <th className="d-none d-lg-table-cell">Timestamp</th>
-                <th className="d-table-cell d-lg-none" colSpan="2">
-                  Hash
-                  <br />
-                  Timestamp
-                </th>
-              </tr>
-            </thead>
-            <tbody>{loadTableBody()}</tbody>
-          </table>
-        </div>
+      setTransactions(data.transactions);
+      setFirstHash(newFirstHash);
+      setFirstTimestamp(newFirstTimestamp);
+      setLastHash(newLastHash);
+      setLastTimestamp(newLastTimestamp);
+      setHasAfter(newHasAfter);
+      setHasBefore(newHasBefore);
+      setLoaded(true);
+    },
+    [pagination]
+  );
+
+  /**
+   * useCallback is needed here because this method is used as a dependency in the useEffect
+   *
+   * Method used to call the props method that will fetch new data
+   * depending on the current query params
+   *
+   * queryParams {Object} Query parameters of the URL
+   */
+  const getData = useCallback(
+    queryParams => {
+      updateData(queryParams.ts, queryParams.hash, queryParams.page).then(
+        data => {
+          handleDataFetched(data, queryParams);
+        },
+        e => {
+          // Error in request
+          console.log(e);
+        }
       );
-    };
+    },
+    [handleDataFetched, updateData]
+  );
 
-    const loadTableBody = () => {
-      return this.state.transactions.map((tx, idx) => {
-        return <TxRow key={tx.tx_id} tx={tx} />;
-      });
+  useEffect(() => {
+    // Handle load history depending on the query params in the URL
+    getData(pagination.obtainQueryParams());
+  }, [location, getData, pagination]);
+
+  useEffect(() => {
+    // Handle new txs in the network to update the list in real time
+    WebSocketHandler.on('network', handleWebsocket);
+
+    return () => {
+      WebSocketHandler.removeListener('network', handleWebsocket);
     };
```

Comment:
[question] Is this unmounting really working? I'm asking this because React has probably a bug when dealing with unmounting when more then 1 `useEffect` is declared in the component. If it is working, all good, otherwise I would solve this issue by declaring this `useEffect` first.

Reply:
I tested and it's working fine when the screen is unmounted. What's this bug that React has with more than one `useEffect`?

Reply:
Maybe this is a thing of React Native. When we use two `useEffect` with `unmounting` callbacks only the the first `unmounting` callback from the first `useEffect` is called, never the second. This case is little different because the first `useEffect` didn't register an `unmounting`, maybe you can make this test latter. However, for this PR it is enough. I'm approving.